src/Validation/HasMatchingPluralization.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace NextApps\PoeditorSync\Validation;
+
+use Closure;
+use Illuminate\Contracts\Validation\ValidationRule;
+use Illuminate\Support\Collection;
+
+class HasMatchingPluralization implements ValidationRule
+{
+    public function validate(string $translationKey, mixed $translations, Closure $fail) : void
+    {
+        $expectedPluralizationParts = $this->getPluralizationParts($translations[$this->getFallbackLocale()]);
+
+        collect($translations)
+            ->except($this->getFallbackLocale())
+            ->each(function (string $text, string $locale) use ($expectedPluralizationParts, $fail) {
+                $pluralizationParts = $this->getPluralizationParts($text);
+
+                if ($expectedPluralizationParts->isNotEmpty() && $pluralizationParts->isEmpty()) {
+                    $fail("Missing pluralization in locale '{$locale}'");
+                } elseif ($expectedPluralizationParts->isEmpty() && $pluralizationParts->isNotEmpty()) {
+                    $fail("Unexpected pluralization in locale '{$locale}'");
+                } elseif ($pluralizationParts->all() !== $expectedPluralizationParts->all()) {
+                    $fail("Invalid pluralization in locale '{$locale}'");
+                }
+            });
+    }
+
+    protected function getPluralizationParts(string $text) : Collection
+    {
+        preg_match_all('/({\d*}*)|(\|)|(\[\d*,(?:\d+|\**)\])/', $text, $matches);
```

Comment:
Misschien moeten we deze regex wijzigen zodat deze ongeveer dezelfde is als de regex die gebruikt wordt in Laravel MessageSelector

```suggestion
        preg_match_all('/(?:(?:^|\|)[\{\[]([^\[\]\{\}]*)[\}\]]|\|)/', $text, $matches);
```
In de MessageSelector is het `/^[\{\[]([^\[\]\{\}]*)[\}\]]/` maar die doen vooraf een split op `|` waardoor de conditie altijd in begin van de string staat. En waardoor ze ook geen rekening houden met een `|` die op zichzelf staat